test/embedded/classes-non-final-method-no-stdlib.swift

```diff
@@ -3,26 +3,26 @@
 // REQUIRES: swift_in_compiler
 
 public class MyClass {
-  func foo<T>(t: T) { } // expected-error {{classes cannot have non-final generic fuctions in embedded Swift}}
-  func bar() { }
+  public func foo<T>(t: T) { } // expected-error {{classes cannot have non-final generic fuctions in embedded Swift}}
+  public func bar() { }
 }
 
-final class C2<Element> {
+public final class C2<Element> {
   // TODO: this shouldn't be a problem because the class is final
-  init<T>(x: T) { } // expected-error {{classes cannot have non-final generic fuctions in embedded Swift}}
+  public init<T>(x: T) { } // expected-error {{classes cannot have non-final generic fuctions in embedded Swift}}
 }
 
-struct S {}
+public struct S {}
 
-func testit2() -> C2<S> {
+public func testit2() -> C2<S> {
   return C2(x: S())
 }
 
 open class C3<X> {
   public func foo<T>(t: T) {} // expected-error {{classes cannot have non-final generic fuctions in embedded Swift}}
 }
 
-func testit3() -> C3<S> {
+public func testit3() -> C3<S> {
```

Comment:
It's because dead-function-elimination runs before PerfDiags.
But I moved the check back to MandatoryPerformanceOptimizations - and reverted the changes in this test file.